backend/plugin/advisor/pg/advisor_naming_fully_qualified.go

```diff
@@ -0,0 +1,300 @@
+				checker.appendAdviceByObjName(tableName.String())
+			}
+		}
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return checker.adviceList, nil
+}
+
+func (checker *FullyQualifiedObjectNameChecker) appendAdviceByObjName(objName string) {
+	if objName == "" {
+		return
+	}
+	re := regexp.MustCompile(`.+\..+`)
+	if !re.MatchString(objName) {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  checker.status,
+			Code:    advisor.NamingNotFullyQualifiedName,
+			Title:   checker.title,
+			Content: fmt.Sprintf("unqualified object name: '%s'", objName),
+			Line:    checker.line,
+		})
+	}
+}
+
+func getFullyQualiufiedObjectName(nodeDef ast.Node) string {
+	sb := strings.Builder{}
+	switch def := nodeDef.(type) {
+	case *ast.TableDef:
+		if def.Database != "" {
+			_, _ = sb.WriteString(def.Database)
+			_, _ = sb.WriteRune('.')
+		}
+		if def.Schema != "" {
+			_, _ = sb.WriteString(def.Schema)
+			_, _ = sb.WriteRune('.')
+		}
+		_, _ = sb.WriteString(def.Name)
+
+	case *ast.IndexDef:
+		if def.Table != nil && def.Table.Schema != "" {
+			_, _ = sb.WriteString(def.Table.Name)
+			_, _ = sb.WriteRune('.')
+		}
+		_, _ = sb.WriteString(def.Name)
+
+	case *ast.SequenceNameDef:
+		if def.Schema != "" {
+			_, _ = sb.WriteString(def.Schema)
+			_, _ = sb.WriteRune('.')
+		}
+		sb.WriteString(def.Name)
+
+	case *ast.ColumnNameDef:
+		if def.Table != nil && def.Table.Name != "" {
+			_, _ = sb.WriteString(def.Table.Name)
+			sb.WriteRune('.')
+		}
+		sb.WriteString(def.ColumnName)
+
+	default:
+	}
+	return sb.String()
+}
+
+// Used for select statement.
+func findAllTables(statement string) []base.ColumnResource {
+	jsonText, err := pgquery.ParseToJSON(statement)
+	if err != nil {
+		return nil
+	}
+
+	var jsonData map[string]any
+	if err := json.Unmarshal([]byte(jsonText), &jsonData); err != nil {
+		return nil
+	}
+
+	resourceArray, err := getRangeVarsFromJSONRecursive(jsonData)
+	if err != nil {
+		return nil
+	}
+
+	return resourceArray
+}
+
+// get table names from Json.
+func getRangeVarsFromJSONRecursive(jsonData map[string]any) ([]base.ColumnResource, error) {
+	var result []base.ColumnResource
+	if jsonData["RangeVar"] != nil {
+		resource := base.ColumnResource{}
+
+		rangeVar, ok := jsonData["RangeVar"].(map[string]any)
+		if !ok {
+			return nil, errors.Errorf("failed to convert range var")
+		}
+		if rangeVar["schemaname"] != nil {
+			schema, ok := rangeVar["schemaname"].(string)
+			if !ok {
+				return nil, errors.Errorf("failed to convert schemaname")
+			}
+			resource.Schema = schema
+		}
+		if rangeVar["relname"] != nil {
+			table, ok := rangeVar["relname"].(string)
+			if !ok {
+				return nil, errors.Errorf("failed to convert relname")
+			}
+			resource.Table = table
+		}
+
+		result = append(result, resource)
```

Comment:
This function will extract both physical and pseudo tables, so we should filter out pseudo tables before we check the table name for this SQL review rule.
Commonly we use Bytebase's metadata to check whether the table we extracted exists or not. If not, we think it's the pseudo table, we no need to check the owner of it.